usr.sbin/arp/arp.c

```diff
@@ -375,7 +375,7 @@ set(int argc, char **argv)
 		}
 	}
 #ifndef WITHOUT_NETLINK
-	return (set_nl(0, dst, &sdl_m, host));
+	return (set_nl(opts.rifindex, dst, &sdl_m, host));
 #else
 	return (set_rtsock(dst, &sdl_m, host));
```

Comment:
I like this approach, I'm wondering if rtsock base API should be change as well...